lib/services/local/headlessauthn_watcher.go

```diff
@@ -140,10 +141,10 @@ func (h *HeadlessAuthenticationWatcher) runWatchLoop(ctx context.Context) {
 			h.Log.Debugf("Attempting to restart watch after waiting %v.", t.Sub(startedWaiting))
 			h.retry.Inc()
 		case <-ctx.Done():
-			h.Log.Debug("Context closed, returning from watch loop.")
+			h.Log.Debugf("Context closed with err: %v. returning from watch loop.", ctx.Err())
 			return
 		case <-h.closed:
-			h.Log.Debug("Watcher closed, returning from watch loop.")
+			h.Log.Debug("Watcher closed. returning from watch loop.")
```

Comment:
nit:

```suggestion
			h.Log.Debug("Watcher closed. Returning from watch loop.")
```

or revert?